### PR TITLE
Provide QuasiQuoters

### LINKED ISSUE #4: Provide quasi-quoters
I'm using this library to transform and modify user-input SQL, so I pass around a lot of AST objects. Some of these are huge and static, so I don't want to copy-paste the entire thing, nor do I want to run the parser at runtime.

It's probably straight-forward to implement, I might give it a shot if this is deemed interesting.

### PR DESCRIPTION
**Pull request by hasufell**
Fixes #4

### CODE REVIEW

File: postgresql-syntax.cabal [hunk 1]
```diff
@@ -47,7 +48,11 @@ library
     parser-combinators >=1.3 && <1.4,
     text >=1 && <3,
     text-builder >=0.6.6.3 && <0.7,
+    template-haskell >= 2.11 && < 2.19,
+    th-compat >= 0.1.3 && < 0.2,
     unordered-containers >=0.2.16 && <0.3
+  ghc-options:        -pgmPcpphs -optP--cpp
+  build-tool-depends: cpphs:cpphs
```
Comment (hasufell) [inline postgresql-syntax.cabal:55]: `/**/` string concatenation in CPP works different between gcc and clang, hence we force the use of cpphs here

**Comment by mgajda** [comment]
I am puzzled why this PR was closed.
Is there a viable alternative that works with variables?

**Comment by hasufell** [comment]
@mgajda not sure, I closed my PRs here since I felt @nikita-volkov does not have capacity or interest attending to them.

**Comment by mgajda** [comment]
Then please reopen, and if Nikita does not care, then we ask `hackage-maintainers`.

**Comment by mgajda** [comment]
Question is whether there is any maintenance and development here.
Since it was originally extracted from `hasql-th`, maybe the whole library should be archived and abandoned.
Then PR may go to [`hasql-th`](https://github.com/nikita-volkov/hasql-th)?

**Comment by mgajda** [comment]
Or alternatively we should have a library on top of `postgresql-syntax` that provides your quasiquoter as alternative to `hasql-th`?

**Comment by nikita-volkov** [comment]
Oops. For some reason I wasn't subscribed to notifications from this repo. If it wasn't for @hasufell  mentioning me recently, I wouldn't have gotten the notification. Hence is the lack of my reaction to either this PR or the issues.

Regarding the PR it looks like it can be packed as a separate library, which simply wraps this one. Why not release it as such?

**Comment by hasufell** [comment]
I don't believe quasiquoters are a reason to create a separate package and I got no interest maintaining one. It will be a mess in case this package gets version bumped. This should be tightly integrated.

**Comment by nikita-volkov** [comment]
Well, I'm not interested in extending my maintenance responsibilities either. I also think that maintenance responsibilities are the decisive practical factor in package separation, the rest is a matter of taste.

**Comment by hasufell** [comment]
> Well, I'm not interested in extending my maintenance responsibilities either. I also think that maintenance responsibilities are the decisive practical factor in package separation, the rest is a matter of taste.

This is only true if you equate maintenance responsibility with package boundaries. There can be separate maintenance responsibilities in one package as well.

**Comment by nikita-volkov** [comment]
I prefer the first option. The second comes at the cost of organisational and communicational overhead which I'm not willing to take here.

**Pull request closed without merging**